app/controllers/CheckYourAnswersController.scala

```diff
@@ -161,19 +163,82 @@ class CheckYourAnswersController @Inject()(
       (request.userAnswers.get(DoYouHaveUTRPage), request.userAnswers.get(RegistrationTypePage),
         request.userAnswers.get(DoYouHaveANationalInsuranceNumberPage)) match {
 
-        case (Some(true), None, None) => createEnrolment(request.userAnswers) // TODO: EIS subscription for business with ID
-        case (Some(false), Some(Individual), Some(true)) => createEnrolment(request.userAnswers) // TODO: EIS subscription for individual with ID
+        case (Some(true), None, None) => createSubscriptionThenEnrolment(request.userAnswers)
+        case (Some(false), Some(Individual), Some(true)) => createSubscriptionThenEnrolment(request.userAnswers)
 
         case (Some(false), _, Some(false) | None) => registrationService.sendRegistration(request.userAnswers) flatMap {
           case Some(response) => response.status match {
-            case OK => createEnrolment(request.userAnswers)
+            case OK => createSubscriptionThenEnrolment(request.userAnswers, Some(response))
             case _ => Future.successful(Redirect(routes.ProblemWithServiceController.onPageLoad()))
           }
           case _ => Future.successful(Redirect(routes.ProblemWithServiceController.onPageLoad()))
         }
         case _ => Future.successful(Redirect(routes.ProblemWithServiceController.onPageLoad()))
       }
+  }
+
+  private def createSubscriptionThenEnrolment(userAnswers: UserAnswers,
+                                              response: Option[HttpResponse] = None)
+                                              (implicit hc: HeaderCarrier): Future[Result] = {
+
+    if (response.isDefined) {
```

Comment:
can we use a case match instead?

Reply:
reponse.map(_.json.validate[PayloadRegistrationWithoutIDResponse]) match {
     case Some(JsSucc...) => 
     .....
    case None => ..
}
then pull out 
createEISSubscription(userAnswersWithSafeID).flatMap {
                userAnswersWithSubscriptionID =>
                  createEnrolment(userAnswersWithSubscriptionID)
              }
into a separate function